File: sentence-library-android/app/src/main/java/com/beomsoo/sentenceapp/OfficialHostPolicy.java
package com.beomsoo.sentenceapp;

import java.net.URI;

public final class OfficialHostPolicy {
  private OfficialHostPolicy() {}

  public static boolean isAllowed(String rawUrl) {
    try {
      URI uri = URI.create(rawUrl);
      if (!"https".equalsIgnoreCase(uri.getScheme())) return false;
      String host = uri.getHost();
      if (host == null) return false;
      host = host.toLowerCase();
      return host.equals("jw.org") || host.endsWith(".jw.org");
    } catch (Exception e) {
      return false;
    }
  }
}

File: sentence-library-v3/app/src/main/java/com/beomsoo/sentencelibrary/sync/OfficialHostPolicy.kt
package com.beomsoo.sentencelibrary.sync

import java.net.URI

object OfficialHostPolicy {
    private val exact=setOf("jw.org","www.jw.org","wol.jw.org","b.jw-cdn.org")
    fun isAllowed(url:String):Boolean {
        return try {
            val uri=URI(url);val host=uri.host?.lowercase() ?: return false
            uri.scheme.equals("https",true) && (host in exact || host.endsWith(".jw.org") || host.endsWith(".jw-cdn.org"))
        } catch(_:Exception){ false }
    }
}
